boards/arm/olimex_stm32_h103/olimex_stm32_h103.dts

```diff
@@ -61,76 +61,17 @@
 	/* usbpre not set: USB clock = 72 / 1.5: 48MHz */
 };
 
-&usart1 {
-	pinctrl-0 = <&usart1_tx_pa9 &usart1_rx_pa10>;
-	pinctrl-names = "default";
-	current-speed = <115200>;
-	status = "okay";
-};
-
```

Comment:
Question here on why removing all these nodes.
Having them enabled allows people to use them more easily in samples/tests.

Note that I don't particularly care on this specific platform, and you were the one who enabled them in first place, so this is really up to you, but in general, on other boards, we have more people asking to enable devices or enabling them themselves than complaining because too much devices are enabled.
On boards made by ST, such change would be a no go. But as I said this is up to you for this specific board.

Reply:
Hi @erwango , thanks for taking the time to look at this.

I think I made a mistake initially when adding support for that board. Some peripherals were arbitrarily enabled by default and without any reason. I don't see why those pins should be configured as a UART and not as a GPIO, for instance.

I understand that this could break someone's application if based on that board, but it should be solvable with an overlay I guess. In addition, this board is identical to [Olimex STM32 H405](https://docs.zephyrproject.org/latest/boards/arm/olimex_stm32_h405/doc/index.html) (just a different MCU) which only has the bare minimum pins enabled.

Note that button, LED and Zephyr console on UART2 are left untouched.

Reply:
Are you sure you want to remove the peripherals from the devicetree? They won't be enabled in an application unless you set `CONFIG_I2C=y`, `CONFIG_SPI=y`, etc. I think you may just want to set `status = disabled` for the unused USART nodes and leave everything else as-is.

Reply:
Hi, sorry for the delay in my reply, I've been away.

Thanks for your suggestion @MaureenHelm, I will leave the peripherals defined with their pinctl configured, but disabled, so it is easier to enable them later.

I will remove the configuration of `adc1` and `timers1` however,  as there is no reason to configure PA1 and PA8 in this way.